src/Storages/ExternalStream/Kafka/Consumer.cpp

```diff
@@ -70,6 +71,9 @@ void Consumer::stopConsume(Topic & topic, Int32 parition)
 
 void Consumer::consumeBatch(Topic & topic, Int32 partition, uint32_t count, int32_t timeout_ms, Consumer::Callback callback, ErrorCallback error_callback) const
 {
+    if (stopped)
+        throw Exception(ErrorCodes::NO_AVAILABLE_KAFKA_CONSUMER, "Cannot consume from stopped consummer");
```

Comment:
The error code is not reflecting a stopped consumer

Reply:
Look like there is still race since a single bool can't eliminate the race but lower the possibility.

Probably you can check existing NativeLog source reader etc which we carefully maintain the dtor sequence and dependencies life cycle.

Reply:
Fixed the error codes. For race, please check my above comment and let me know if it makes sense.